src/addon/info.hpp

```diff
@@ -22,10 +22,67 @@
 #include <set>
 #include <map>
 
+struct addon_info_translation;
 struct addon_info;
 class config;
 typedef std::map<std::string, addon_info> addons_list;
 
+struct addon_info_translation
+{
+	static addon_info_translation invalid;
+
+	std::string language;
+	bool supported;
+	std::string title;
+	std::string description;
+
+	addon_info_translation()
+		: language()
+		, supported(true)
+		, title()
+		, description()
+	{}
+
+	addon_info_translation(std::string lang, bool sup, std::string titl, std::string desc)
+		: language(lang)
+		, supported(sup)
+		, title(titl)
+		, description(desc)
+	{
+	}
+
+	explicit addon_info_translation(const config& cfg)
+		: language()
+		, supported(true)
+		, title()
+		, description()
+	{
+		this->read(cfg);
+	}
+
+	addon_info_translation(const addon_info_translation&) = default;
+
+	addon_info_translation& operator=(const addon_info_translation& o)
+	{
+		if(this != &o) {
+			this->language = o.language;
+			this->supported = o.supported;
+			this->title = o.title;
+			this->description = o.description;
+		}
+		return *this;
+	}
+
+	void read(const config& cfg);
+
+	void write(config& cfg) const;
+
+	bool valid()
+	{
+		return title != "invalid_addon!";
+	}
```

Comment:
There's technically nothing stopping me from titling my add-on `invalid_addon!` in either the original language or translations, is there?

Reply:
Well, technically yes. Would you suggest adding a filter on the server side or check whether it's invalid in some other way?

Reply:
AFAICT the best validity criterion here (as in the only one that doesn't require contorting the implementation of anything else around it) really is checking if the translated title is empty.

Reply:
Ok, very well.